### DISCUSSION TITLE
temp2 file should not be copied if final dir on same volume

### DISCUSSION THREAD

**Discussion by cross** [Chia-Network/chia-blockchain#8846] [Ideas]
In the less common case where temp dir, or at least temp2, is on the same filesystem as the final dir, rename should be used instead of copy.   Using chia 1.2.7 on linux, I started a plot as follows:
```
Starting plotting progress into temporary dirs: /data/chia/tmp and /data/chia/tmp
ID: [...]
Plot size is: 34
[...]
Final Directory is: /data/chia/plots/k34
Using 16 threads of stripe size 65536
```

And on my system, `/data` is a large RAID array, so despite the spindle contention, it's all on the same device.  But at the end of this plot, I see:
```
Approximate working space used (without final file): 1176.562 GiB
Final File size: 429.934 GiB
Total time = 135175.666 seconds. CPU (145.320%) Mon Oct 11 02:07:45 2021
Copied final file from "/data/chia/tmp/plot-k34-2021-10-09-12-34-<keyval>.plot.2.tmp" to "/data/chia/plots/k34/plot-k34-2021-10-09-12-34-<keyval>.plot.2.tmp"
Copy time = 1952.410 seconds. CPU (37.220%) Mon Oct 11 02:40:39 2021
Removed temp2 file "/data/chia/tmp/plot-k34-2021-10-09-12-34-<keyval>.plot.2.tmp"? 1
Renamed final file from "/data/chia/plots/k34/plot-k34-2021-10-09-12-34-<keyval>.plot.2.tmp" to "/data/chia/plots/k34/plot-k34-2021-10-09-12-34-<keyval>.plot"
```

Unless I'm mistaken, the 1952 seconds spent copying the final file should've been a `rename` operation, which would also eliminate the removal of the temp2 file.

**Comment by BrandtH22** [comment]
@cross , since this ticket was opened a number of years ago without further comments it seems this issue has been resolved. 
We will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions or comments.

Note the temp2 drive is not required so just temp1 and the destination can be used negating this issue.

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
